### DISCUSSION TITLE
Fatal error: Received an output without requesting demand

### DISCUSSION THREAD

**Discussion by wtholliday** [netreconlab/Parse-Swift#178] [Q&A]
I'm getting this error:

<img width="1035" alt="image" src="https://github.com/user-attachments/assets/fb2ecf8e-014a-495a-b4f1-401f351a18ed">

Seems like there's a bug in Parse Swift with respect to how it uses Combine. Here's some more info I found: https://forums.swift.org/t/asyncpublisher-causes-crash-in-rather-simple-situation/56574/10

Anyone have some ideas? Thanks!

**Comment by cbaker6** [comment]
Can you add more context for what your code is doing when this happens? For example, the relevant code for your `ParseObject`, live query, the code for your `View` (if you used one), what type of data was sent to the server to trigger this? Can you also provide the scrubbed version of `newValue` and `event` at the time of the crash? Can you also provide the info from the stack trace of the crash?

**Reply by wtholliday** [reply]
Thanks for your help @cbaker6 !

Here's `newValue`:

<img width="421" alt="image" src="https://github.com/user-attachments/assets/defef3ec-84b5-4dfc-966c-11e3e07e1846">

Here's the stack trace:

<img width="419" alt="image" src="https://github.com/user-attachments/assets/bd491bda-5861-4d25-b437-b3030bbc404f">


Here's how I'm setting up the LiveQuery:

```
eventUpdateTask = Task.detached { [weak self] in
            // Setup Event subscription
            let eventSubscription = try await eventQuery.subscribe()

            for await _ in eventSubscription.objectWillChange.values {
                await self?.handleEventUpdate(eventID: eventID)
            }
        }
```

Which does differ from the LiveQuery examples in the playground so maybe I'm misusing the API?

Thanks again! :)

**Reply by cbaker6** [reply]
It's possible that your current use is causing the issue, `[weak self]` could be deallocated inside the closure. I've never seen the error you are getting, but I've also never used a code block similar to yours.

**Reply by wtholliday** [reply]
Rewriting using the playground example as a guide seems to fix it. I think originally I was trying to avoid this warning about `SubscriptionCallback` not being `Sendable`:

<img width="1258" alt="image" src="https://github.com/user-attachments/assets/d2c7ff93-6aff-4254-960d-71ac8f55abe0">

**Reply by cbaker6** [reply]
The warning makes sense, `ParseObjects` are `Sendable` since [v5.10.0](https://github.com/netreconlab/Parse-Swift/releases/tag/5.10.0), but a `Subscription` is an `ObservableObject` which has to be a reference type/class, so it is definitely not `Sendable`. Your current code can easily cross over thread bounderies, but if you are tying your subscription to a `View` or using the `ViewModel` of the query, you should be okay
